### PR TITLE
GH-327 Update repo resource documentations to include missing attributes

### LINKED ISSUE #327: project key and project environment arguments support for local docker v2 repository.
https://registry.terraform.io/providers/jfrog/artifactory/latest/docs/resources/artifactory_local_docker_v2_repository

Can we have all the arguments from the Jfrog API configurable especially project key and environment?

### PR DESCRIPTION
**Pull request by alexhung**
Closes #327 

I tried not to repeat the documentation too much by referencing the 'generic' repo documentation. But since we don't have this for remote repos yet, I have to duplicate the doc for each remote repo type 😞

### CODE REVIEW

File: docs/resources/artifactory_local_bower_repository.md [hunk 1]
```diff
@@ -18,4 +18,4 @@ Arguments have a one to one mapping with the [JFrog API](https://www.jfrog.com/c
 * `description` - (Optional)
 * `notes` - (Optional)
 
-Arguments for Bower repository type closely matches with arguments for Generic repository type. 
\ No newline at end of file
+Arguments for Bower repository type closely matches with arguments for Generic repository type.
```
Comment (chb0github) [inline docs/resources/artifactory_local_bower_repository.md:21]: Closely match..

File: docs/resources/artifactory_local_generic_repository.md [hunk 2]
```diff
@@ -14,9 +14,17 @@ resource "artifactory_local_generic_repository" "terraform-local-test-generic-re
...
+* `key` - (Required) - A mandatory identifier for the repository that must be unique. It cannot begin with a number or contain spaces or special characters.
 * `description` - (Optional)
 * `notes` - (Optional)
 * `project_key` - (Optional) Project key for assigning this repository to. When assigning repository to a project, repository key must be prefixed with project key, separated by a dash.
 * `project_environments` - (Optional) Project environment for assigning this repository to. Allow values: "DEV" or "PROD"
-* `xray_index` - (Optional, Default: false)  Enable Indexing In Xray. Repository will be indexed with the default retention period. You will be able to change it via Xray settings.
+* `includes_pattern` - (Optional) List of artifact patterns to include when evaluating artifact requests in the form of x/y/**/z/*. When used, only artifacts matching one of the include patterns are served. By default, all artifacts are included (**/*).
+* `excludes_pattern` - (Optional) List of artifact patterns to exclude when evaluating artifact requests, in the form of x/y/**/z/*. By default no artifacts are excluded.
+* `repo_layout_ref` - (Optional)
```
Comment (chb0github) [inline docs/resources/artifactory_local_generic_repository.md:24]: No comment on this one? It's the named repo layout

File: docs/resources/artifactory_local_generic_repository.md [hunk 3]
```diff
@@ -14,9 +14,17 @@ resource "artifactory_local_generic_repository" "terraform-local-test-generic-re
...
+* `blacked_out` - (Optional, Default: false)
+* `xray_index` - (Optional, Default: false) Enable Indexing In Xray. Repository will be indexed with the default retention period. You will be able to change it via Xray settings.
+* `priority_resolution` - (Optional, Default: false) Setting repositories with priority will cause metadata to be merged only from repositories set with this field
+* `property_sets` - (Optional)
```
Comment (chb0github) [inline docs/resources/artifactory_local_generic_repository.md:28]: I have no idea what this means

File: docs/resources/artifactory_remote_cargo_repository.md [hunk 4]
```diff
@@ -25,13 +25,45 @@ you must set the base url at: `http://${host}/ui/admin/configuration/general`
...
+* `hard_fail` - (Optional)
+* `offline` - (Optional) If set, Artifactory does not try to fetch remote artifacts. Only locally-cached artifacts are retrieved.
+* `blacked_out` - (Optional) (A.K.A 'Ignore Repository' on the UI) When set, the repository or its local cache do not participate in artifact resolution.
 * `anonymous_access` - (Required) - Cargo client does not send credentials when performing download and search for crates. Enable this to allow anonymous access to these resources (only), note that this will override the security anonymous access option.
+* `xray_index` - (Optional, Default: false)  Enable Indexing In Xray. Repository will be indexed with the default retention period. You will be able to change it via Xray settings.
+* `store_artifacts_locally` - (Optional) When set, the repository should store cached artifacts locally. When not set, artifacts are not stored locally, and direct repository-to-client streaming is used. This can be useful for multi-server setups over a high-speed LAN, with one Artifactory caching certain data on central storage, and streaming it directly to satellite pass-though Artifactory servers.
+* `socket_timeout_millis` - (Optional)
```
Comment (chb0github) [inline docs/resources/artifactory_remote_cargo_repository.md:46]: No comment?

**Review by chb0github — COMMENTED** [review]
You repeat the same, very minor grammar mistake. And, you have plenty of attributes without comment.

Though, i commend you on finding as many as you have

File: docs/resources/artifactory_remote_cargo_repository.md (hunk 4 above)
Comment (alexhung) [reply docs/resources/artifactory_remote_cargo_repository.md:46]: Found it in the UI.

**Review by dasmanas — APPROVED** [review]
LGTM

**Comment by maheshjfrog** [comment]
Nice changes. Looks fine.
For the descriptions (for Optional ones) that got added new ... Default values are missing.  Approving.

**Review by maheshjfrog — APPROVED** [review]

**Review by dasmanas — APPROVED** [review]

**Review by chb0github — APPROVED** [review]

**Pull request merged**